tests/test_kernels.py

```diff
@@ -0,0 +1,89 @@
+"""Kernel tests."""
+
+from typing import Any
+
+import numpy as np
+import torch
+from attrs import asdict, has
+from hypothesis import given
+
+from baybe.kernels.base import Kernel
+from baybe.kernels.composite import AdditiveKernel, ProductKernel, ScaleKernel
+
+from .hypothesis_strategies.kernels import kernels
+
+# TODO: Consider deprecating these attribute names to avoid inconsistencies
+_RENAME_DICT: dict[str, str] = {
+    "base_kernels": "kernels",
+}
+"""Dictionary for resolving name differences between BayBE and GPyTorch attributes."""
+
+
+def validate_kernel_components(obj: Any, mapped: Any, **kwargs) -> None:
+    """Validate that all kernel components are translated correctly.
+
+    Args:
+        obj: An object occurring as part of a BayBE kernel.
+        mapped: The corresponding object in the translated GPyTorch kernel.
+        **kwargs: Optional kernel arguments that were passed to the GPyTorch kernel.
+    """
+    # Assert that the kernel kwargs are correctly mapped
+    # TODO: Consider introducing intermediate kernel classes or class variables
+    #   to differentiate basic kernels from composite ones
+    if isinstance(obj, Kernel) and not isinstance(
+        obj, (ScaleKernel, AdditiveKernel, ProductKernel)
+    ):
+        for k, v in kwargs.items():
```

Comment:
I'm confused here - which arguments are tested by this? Can you give some examples?

Reply:
it's exactly the stuff that we pass to `to_gpytorch` (i.e. everything that is not an attribute): `ard_num_dims, batch_shape, active_dims`